terraform/iam_user.tf

```diff
@@ -9,42 +13,22 @@ resource "aws_iam_access_key" "key" {
   user = aws_iam_user.user.name
 }
 
-# The S3 bucket of interest
-data "aws_s3_bucket" "bucket" {
-  bucket = var.bucket_name
-}
-
-# IAM policy document that allows reading from a specific S3 bucket.
-# This will be applied to the IAM user we are creating.
-data "aws_iam_policy_document" "s3_doc" {
-  statement {
-    effect = "Allow"
-
-    actions = [
-      "s3:ListBucket",
-    ]
-
-    resources = [
-      data.aws_s3_bucket.bucket.arn,
-    ]
-  }
-
-  statement {
-    effect = "Allow"
-
-    actions = [
-      "s3:GetObject",
-    ]
-
-    resources = [
-      "${data.aws_s3_bucket.bucket.arn}/*",
-    ]
-  }
-}
-
-# The S3 policy for our IAM user that lets the user read from a
-# particular bucket.
-resource "aws_iam_user_policy" "s3_policy" {
-  user   = aws_iam_user.user.id
-  policy = data.aws_iam_policy_document.s3_doc.json
+module "bucket_access" {
+  source = "github.com/cisagov/s3-read-role-tf-module"
+
+  account_ids = [data.aws_caller_identity.current.account_id]
+  entity_name = aws_iam_user.user.name
+  # When the user is not yet created, Terraform does not correctly pick up
+  # the user's name from aws_iam_user.user.name when passing it in on
+  # the next line, so we fall back to using local.test_user_name.
+  iam_usernames = [local.test_user_name]
```

Comment:
I don't get why you have to use `local.test_user_name` but you can use `aws_iam_user.user.name` on the lines above and below.  Shouldn't using `aws_iam_user.user.name` _force_ the creation of the user before this module?

Reply:
I don't get it either.  I agree that it should do that, but it doesn't.  Chalk it up to the mysteries of the Terraform dependency graph.

Reply:
Go nuts man!

Reply:
See commit 3586161.